### PR TITLE
Implement missing Account methods

### LINKED ISSUE #405: Implement Accounts methods and entities
Mastodon’s [Account API](https://docs.joinmastodon.org/methods/accounts/) allows a user to interact with everything account-related, such as registering an account, verifying and updating credentials, getting someone else’s account and their statuses, followers, following, featured tags, as well as following and unfollowing plus a few more. 

The goal of this task is to implement the following endpoints of this API in BigBone:

> [!NOTE]
> Some of them have already been implemented but are missing some properties, others are entirely missing, some are fully supported.
> Ideally, you can start with `AccountMethods.kt` and work your way out from there.

## Entirely missing

* Get account’s featured tags
* Get lists containing this account
* Remove account from followers
* Feature account on your profile
* Unfeature account from profile
* Set private note on profile
* Find familiar followers
* Lookup account ID from Webfinger address

## Missing parameters or return values are missing properties:

Already done in #411

* ~~Register an account~~
* ~~Verify account credentials~~
* ~~Update account credentials~~
* ~~Get account’s statuses~~
* ~~Follow account~~
* ~~Mute account~~
* ~~Check relationships to other accounts~~
* ~~Search for matching accounts~~

> [!TIP]
> You can find the current status of already merged and thus available methods in [API coverage page of the Accounts methods](https://bigbone.social/api-coverage/accounts.html).

The Accounts API as documented by Mastodon can be found here:
[docs.joinmastodon.org/methods/accounts](https://docs.joinmastodon.org/methods/accounts/)

---

Please consult the following documents to make sure your contribution meets our projects criteria:

[Code of Conduct](https://github.com/andregasser/bigbone/blob/master/CODE_OF_CONDUCT.md?rgh-link-date=2023-11-03T07%3A22%3A10Z)

[Code Contribution Guidelines](https://github.com/andregasser/bigbone/blob/master/CONTRIBUTING.md?rgh-link-date=2023-11-03T07%3A22%3A10Z)

### PR DESCRIPTION
**Pull request by PattaFeuFeu**
# Description

Part 2 after #411. This time with all methods that were entirely missing previously.

Nothing special going on here, just adding some new methods.

Closes #405 

# Type of Change

- New feature
- Documentation

# Breaking Changes

- None

# How Has This Been Tested?

New unit tests for every method

# Mandatory Checklist

- [x] I ran `gradle check` and there were no errors reported
- [x] I have performed a self-review of my code
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] All tests pass locally with my changes
- [x] I have added KDoc documentation to all public methods

# Optional checks

- [x] In case you worked on a new feature: Did you also implement the reactive endpoint (bigbone-rx)?
- [x] In case you added new `*Methods` classes: Did you also reference it in the `MastodonClient` main class?
- [x] Did you also update the documentation in the `/docs` folder (e.g. API Coverage page)?

### CODE REVIEW

File: bigbone-rx/src/main/kotlin/social/bigbone/rx/RxAccountMethods.kt [hunk 1]
```diff
@@ -260,6 +299,45 @@ class RxAccountMethods(client: MastodonClient) {
...
+
+    /**
+     * Sets a private note on a user.
+     *
+     * @param accountId The ID of the [Account] in the database.
+     * @param privateNote The comment to be set on that user. Provide empty string or null to clear currently set note.
+     *
+     * @see <a href="https://docs.joinmastodon.org/methods/accounts/#note">Mastodon API documentation: methods/accounts/#note</a>
+     */
+    fun setPrivateNotOnProfile(
```
Comment (andregasser) [inline bigbone-rx/src/main/kotlin/social/bigbone/rx/RxAccountMethods.kt:334]: Typo: Should be `setPrivateNoteOnProfile`

Comment (PattaFeuFeu) [reply bigbone-rx/src/main/kotlin/social/bigbone/rx/RxAccountMethods.kt:334]: Whoopsie! ✍️

Comment (PattaFeuFeu) [reply bigbone-rx/src/main/kotlin/social/bigbone/rx/RxAccountMethods.kt]: ✅

**Review by andregasser — APPROVED** [review]
All good! Thank you very much!

**Pull request merged**